pythia/models/butd.py

```diff
@@ -119,28 +119,37 @@ def forward(self, sample_list):
             dtype=torch.float,
         )
 
-        decoder = registry.get_decoder_class(self.config["inference"]["type"])(self.vocab, self.config)
+        if self.config["inference"]["type"] in ["beam_search", "nucleus_sampling"]:
```

Comment:
We should maybe add a function in registry to get registered decoders and similar function for other methods.